test/util_spec.js

```diff
@@ -0,0 +1,626 @@
+import {expect} from 'chai';
+
+import {
+  addTagToFilter,
+  capitalize,
+  cleanseTags,
+  filterCourses,
+  filterLessons,
+  fixNonArrayTagList,
+  lessonHasTags,
+  mergeObjects,
+  removeTagFromFilter
+} from '../src/util';
+
+describe('util', () => {
+
+  describe('capitalize', () => {
+
+    it('makes first letter in a string capital', () => {
+      const text = 'a random string';
+
+      expect(capitalize(text)).to.equal('A random string');
+    });
+
+    it('is pure', () => {
```

Comment:
Først må jeg bare si at jeg er kjempeimponert over hvor fort du plukker opp kommentarene og forbedrer koden!

Og veldig kult at du legger inn disse "is pure"-testene! Jeg har et lite tips her, for å spare litt arbeid, og det går ut på at det finnes en måte vi kan teste at funksjonene er pure samtidig som vi gjør de andre testene.

Argumenter til funksjoner kan sendes inn "by value" eller "by reference". Når vi sender inn "by value", så er det ikke mulig å endre på original-verdiene, så da trenger man ikke være bekymret for at funksjonen skal være "un-pure". Dette gjelder tall og booleans.

Når man sender inn objekter og arrays (som er "by reference"), så kan man bruke Object.freeze() (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) for å fryse dem før man sender dem inn; hvis de da blir endret, så får man beskjed. I praksis er det bedre å bruke f.eks. deep-freeze (https://github.com/substack/deep-freeze), som rekursivt går gjennom og fryser objekter og arrays.

Når det gjelder strings, så er disse gjort immutable i javascript, så de oppfører seg som om de ble sendt "by value"; derfor trenger man heller ikke fryse disse.

Dette betyr at vi ikke trenger "is pure" testene; når man sender inn strings, tall og booleans er det slik automatisk, og man kan bruke deep-freeze før man sender inn arrays og objects.

Se forøvrig
http://docstore.mik.ua/orelly/webprog/jscript/ch11_02.htm